src/Mjolnir.ts

```diff
@@ -473,31 +497,37 @@ export class Mjolnir {
         );
     }
 
-    public async enableProtection(protectionName: string, persist = true): Promise<any> {
-        const definition = PROTECTIONS[protectionName];
-        if (!definition) throw new Error("Failed to find protection by name: " + protectionName);
+    /*
+     * Given a protection object; add it to our list of protections, set whether it is enabled
+     * and update its settings with any saved non-default values.
+     *
+     * @param protection The protection object we want to register
+     */
+    public async registerProtection(protection: IProtection): Promise<any> {
+        this.protections.set(protection.name, protection)
 
-        const protection = definition.factory();
-        this.protections.push(protection);
+        let protections: { enabled: string[] } | null = null;
```

Comment:
Nit: you have three variables `protection`, `protections` and `this.protections`. Do you think you could rename one of them? :)